test/Table_Tests/src/Common_Table_Operations/Derived_Columns_Spec.enso

```diff
@@ -229,6 +229,10 @@ spec setup =
             t2 = t.set (Column_Operation.Add "prefix" (Column_Ref.Name "X"))
             t3 = t2.set (Column_Operation.Add "prefix" "X")
 
-            t3.column_names . should_equal ["X", "['prefix'] + [X]", "['prefix'] + 'X'"]
-            t3.at "['prefix'] + [X]" . to_vector . should_equal ["prefixa", "prefixb", "prefixc"]
-            t3.at "['prefix'] + 'X'" . to_vector . should_equal ["prefixX", "prefixX", "prefixX"]
+            t3.column_names . should_equal ["X", "[prefix] + [X]", "[prefix] + 'X'"]
+            t3.at "[prefix] + [X]" . to_vector . should_equal ["prefixa", "prefixb", "prefixc"]
+            t3.at "[prefix] + 'X'" . to_vector . should_equal ["prefixX", "prefixX", "prefixX"]
+
+        Test.specify "Should not disambiguate if set_mode is Update" <|
+            t = table_builder [["X", [1, 2, 3]]]
+            t2 = t.set (Column_Operation.Add (Column_Ref.Name "X") 1) set_mode=Set_Mode.Update
```

Comment:
I'm glad we are testing this, but this test only checks that `set` does not panic.

We need to check the contents of `t2`:
```
t2.column_names . should_equal ["X"]
t2.at "X" . to_vector . should_equal [2, 3, 4]
```

Reply:
Done